src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -380,44 +380,71 @@ internal static bool HasThisConstructorInitializer(this MethodSymbol method)
                     ConstructorDeclarationSyntax constructorSyntax = sourceMethod.SyntaxNode as ConstructorDeclarationSyntax;
                     if (constructorSyntax != null)
                     {
-                        ConstructorInitializerSyntax initializerSyntax = constructorSyntax.Initializer;
-                        if (initializerSyntax != null)
+                        if (constructorSyntax.Initializer?.Kind() == SyntaxKind.ThisConstructorInitializer)
                         {
-                            return initializerSyntax.Kind() == SyntaxKind.ThisConstructorInitializer;
+                            initializerSyntax = constructorSyntax.Initializer;
+                            return true;
                         }
                     }
                 }
             }
 
+            initializerSyntax = null;
             return false;
         }
 
         internal static bool IncludeFieldInitializersInBody(this MethodSymbol methodSymbol)
         {
             return methodSymbol.IsConstructor()
-                && !methodSymbol.HasThisConstructorInitializer()
+                && !(methodSymbol.HasThisConstructorInitializer(out var initializerSyntax) && !isDefaultValueTypeConstructor(methodSymbol.ContainingType, initializerSyntax))
                 && !(methodSymbol is SynthesizedRecordCopyCtor) // A record copy constructor is special, regular initializers are not supposed to be executed by it.
                 && !Binder.IsUserDefinedRecordCopyConstructor(methodSymbol);
+
+            // A struct constructor that calls ": this()" will need to include field initializers if the
```

Comment:
The spec may need to be updated. It currently says "When an instance constructor has no constructor initializer, ...". But the default value type constructor doesn't  count.

Reply:
Updated spec in [https://github.com/dotnet/csharplang/pull/4778](https://github.com/dotnet/csharplang/pull/4778).